tests/cxx/unit_tests/xyz_to_mol.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024 NWChemEx-Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../test_nux.hpp"
+#include <nux/nux.hpp>
+
+TEST_CASE("XYZToMolecule") {
+    pluginplay::ModuleManager mm;
+    nux::load_modules(mm);
+
+    auto make_atoms = [=](auto&& Z) {
+      double h_mass = 1837.4260218693814;
+      return simde::type::atom{"H", 1, h_mass, 0.0, 0.0, 0.0};
+    };
```

Comment:
```suggestion
    auto make_atoms = [=](auto&& Z) {
      REQUIRE(Z == 1);
      double h_mass = 1837.4260218693814;
      return simde::type::atom{"H", 1, h_mass, 0.0, 0.0, 0.0};
    };
```
It's not required, but you can use the lambda modules to check that the new module is passing the expected inputs to the submodules.